### PR TITLE
[Graphbolt] Define API for Neighbor sampling.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by peizhou001 — APPROVED** [review]

File: graphbolt/src/csc_sampling_graph.cc [hunk 1]
```diff
@@ -121,6 +121,15 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
           : torch::nullopt);
 }
 
+c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
```
Comment (Rhett-Ying) [inline graphbolt/src/csc_sampling_graph.cc:124]: no declaration in header file?

File: tests/python/pytorch/graphbolt/test_csc_sampling_graph.py [hunk 2]
```diff
@@ -385,6 +385,42 @@ def test_in_subgraph_heterogeneous():
...
+
+    # Construct CSCSamplingGraph.
+    graph = gb.from_csc(indptr, indices)
+
+    # Extract in subgraph.
+    nodes = torch.LongTensor([1, 3, 4])
+    subgraph = graph.sample_neighbors(nodes)
+
+    # Verify in subgraph.
+    assert torch.equal(subgraph.indptr, torch.LongTensor([0, 0, 0]))
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_csc_sampling_graph.py:416]: `[0, 0, 0, 0]`?

File: tests/python/pytorch/graphbolt/test_csc_sampling_graph.py [hunk 3]
```diff
@@ -385,6 +385,42 @@ def test_in_subgraph_heterogeneous():
...
+    num_edges = 12
+    indptr = torch.LongTensor([0, 3, 5, 7, 9, 12])
+    indices = torch.LongTensor([0, 1, 4, 2, 3, 0, 1, 1, 2, 0, 3, 4])
+    assert indptr[-1] == num_edges
+    assert indptr[-1] == len(indices)
+
+    # Construct CSCSamplingGraph.
+    graph = gb.from_csc(indptr, indices)
+
+    # Extract in subgraph.
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_csc_sampling_graph.py:411]: `in subgraph`? a bit confusing and alike `graph.in_subgraph()` though the output graphs are same if `fantout=-1` for `sample_neighbors()`

**Comment by frozenbugs** [comment]
taken by https://github.com/dmlc/dgl/pull/5766.

**Pull request closed without merging**